# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/ServiceWorkerGlobalScope.webidl
# Commit: 2fc9a3d0ad8b
# Full Hash: 2fc9a3d0ad8bdbce325df78fcd731e925245afaf
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-12 09:23:07
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 1 - CookieStore API - WebIDL, r=smaug,edgul
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215144
# ==============================================================================

diff -r 2d598be5ec67 -r 2fc9a3d0ad8b dom/webidl/ServiceWorkerGlobalScope.webidl
--- a/dom/webidl/ServiceWorkerGlobalScope.webidl	Wed Sep 11 16:09:37 2024 +0000
+++ b/dom/webidl/ServiceWorkerGlobalScope.webidl	Wed Sep 11 16:26:54 2024 +0000
@@ -7,6 +7,7 @@
  * http://slightlyoff.github.io/ServiceWorker/spec/service_worker/index.html
  * http://w3c.github.io/push-api/
  * https://notifications.spec.whatwg.org/
+ * https://wicg.github.io/cookie-store/#ServiceWorkerGlobalScope
  *
  * You are granted a license to use, reproduce and create derivative works of
  * this document.
@@ -47,3 +48,13 @@
 // Mixin the WebExtensions API globals (the actual properties are only available to
 // extension service workers, locked behind a Func="extensions::ExtensionAPIAllowed" annotation).
 ServiceWorkerGlobalScope includes ExtensionGlobalsMixin;
+
+// https://wicg.github.io/cookie-store/#ServiceWorkerGlobalScope
+partial interface ServiceWorkerGlobalScope {
+  [SameObject, Pref="dom.cookieStore.enabled"] readonly attribute CookieStore cookieStore;
+
+/* Bug 1475599 - We decide to do not implement the entire cookie-store spec.
+ * Instead, we implement only the subset that is compatible with document.cookie
+ attribute EventHandler oncookiechange;
+ */
+};